python/cuml/common/memory_utils.py

```diff
@@ -255,40 +257,57 @@ def _rmm_cupy6_array_like(ary, order):
     return arr
 
 
-def _strides_to_order(strides, dtype):
+def _strides_to_order(strides, shape, dtype):
     # cuda array interface specification
     if strides is None:
         return 'C'
-    if strides[0] == dtype.itemsize or len(strides) == 1:
+
+    itemsize = cp.dtype(dtype).itemsize
+    shape = list(shape)
+    strides = list(strides)
```

Comment:
I don't think there is a need for these copies and you can also just use the sliced view of those arrays directly.